library/src/main/java/kwasm/format/text/Type.kt

```diff
@@ -62,25 +69,55 @@ sealed class Type(
     class Result(
         sequence: CharSequence,
         context: ParseContext? = null
-    ) : Type(sequence, context) {
+    ) : Type<Unit>(sequence, context) {
         override fun parseValue() {
             TODO("not implemented")
         }
     }
 
+    /**
+     * ```
+     *   limits ::=  n:u32        => {min n, max ϵ}
+     *           |   n:u32  m:u32 => {min n, max m}
+     * ```
+     */
     class Limits(
         sequence: CharSequence,
         context: ParseContext? = null
-    ) : Type(sequence, context) {
-        override fun parseValue() {
-            TODO("not implemented")
+    ) : Type<Pair<UInt, UInt>>(sequence, context) {
```

Comment:
Instead of `Pair<UInt, UInt>`, consider defining a `Limit` type in `kwasm.ast` and using that

Reply:
Done